src/Microsoft.AspNet.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -35,8 +35,16 @@ public static class TagHelperOutputExtensions
             {
                 // We look for the original attribute so we can restore the exact attribute name the user typed.
                 // Approach also ignores changes made to tagHelperOutput[attributeName].
-                var entry = context.AllAttributes.First(
+                var entry = context.AllAttributes.FirstOrDefault(
                     attribute => attribute.Key.Equals(attributeName, StringComparison.OrdinalIgnoreCase));
+
+                if (entry.Equals(default(KeyValuePair<string, object>)))
+                {
+                    throw new ArgumentException(
+                        string.Format("The specified attribute '{0}' does not exists in the attributes of the TagHelperContext.", attributeName),
```

Comment:
How about:

`The attribute '{0}' does not exist in the {1}.`

And then pass in `nameof(TagHelperContext)` for the `{1}`.

Also, can you put the message in the RESX file (look at the other examples where exception messages are used, e.g. https://github.com/henkmollema/Mvc/blob/fix-1586/src/Microsoft.AspNet.Mvc.TagHelpers/SelectTagHelper.cs#L66-L70

After editing the RESX file, I think you need to run `build resx` from the command line to regenerate the `.designer.cs` file.